modules/library/src/main/scala/zio/elasticsearch/query/Queries.scala

```diff
@@ -109,6 +110,101 @@ private[elasticsearch] final case class Exists[S](field: String) extends ExistsQ
     Obj("exists" -> Obj("field" -> fieldPath.foldRight(field)(_ + "." + _).toJson))
 }
 
+sealed trait GeoDistanceQuery[S] extends ElasticQuery[S] {
+
+  /**
+   * sets the `distance` parameter for the [[zio.elasticsearch.query.GeoDistanceQuery]]
+   *
+   * `Distance` represents the radius of the circle centred on the specified location. Points which fall into this
+   * circle are considered to be matches. The distance can be specified in various units. See
+   * [[zio.elasticsearch.query.DistanceUnit]].
+   *
+   * @param value
+   *   the [[Double]] value for distance parameter
+   * @param unit
+   *   the [[zio.elasticsearch.query.DistanceUnit]] in which we want to represent the distance
+   * @return
+   *   a new instance of the [[zio.elasticsearch.query.GeoDistanceQuery]] with the score value set.
+   */
+  def distance(value: Double, unit: DistanceUnit): GeoDistanceQuery[S]
+
+  /**
+   * sets the `distanceType` parameter for the [[zio.elasticsearch.query.GeoDistanceQuery]]
+   *
+   * Defines how to compute the distance. Can either be [[zio.elasticsearch.query.DistanceType.Arc]] (default), or
```

Comment:
Check [this](https://github.com/lambdaworks/zio-elasticsearch/blob/main/modules/library/src/main/scala/zio/elasticsearch/ElasticSort.scala#L71) and you can use a similar format.